tests/FSharp.Compiler.ComponentTests/Language/ConstraintIntersectionTests.fs

```diff
@@ -0,0 +1,62 @@
+﻿module Language.ConstraintIntersectionTests
```

Comment:
One more thought regarding "NicePrint"ing this:

Can you please add this whole sample to become part of SigGenerationRoundTripTests ?
That way, we get to see:
- If the auto-generated .fsi for it works
- If the auto-generated .fsi is back-compilable with this .fs original

Also related to NicePrint-ing:
When function like 'z' here below is used and VS does a mouseover with a tooltip - I assume it renders that using the "and" syntax, right?

Reply:
> When function like 'z' here below is used and VS does a mouseover with a tooltip - I assume it renders that using the "and" syntax, right?

That's right. This quirk (wouldn't necessarily call it a problem or a bug) can already be observed today.

![devenv_RkxCaK9u2V](https://github.com/dotnet/fsharp/assets/5063478/cd02777a-c366-45dc-92ef-a500ca079606)

Not only is the constraint format different, but the `seq` is rendered using the postfix style.

On the other hand, class tooltips do preserve the chosen typar declaration style.

![image](https://github.com/dotnet/fsharp/assets/5063478/097043c1-eb03-42d1-a85c-e1bbebeba1c4)

I'm fine with leaving it like this, not least so as to avoid more, possibly quite complicated, work.

Reply:
Agree, it is not a problem since it is semantically the same.
And the sig-generation roundtrip works, thanks for adding that test!